src/language/concurrency.md

```diff
@@ -16,90 +19,99 @@ prevpage:
   }
 </style>
 
-Dart supports concurrent programming with async-await, isolates, and
-classes such as `Future` and `Stream`.
-This page gives an overview of async-await, `Future`, and `Stream`,
-but it's mostly about isolates.
-
-Within an app, all Dart code runs in an _isolate._
-Each Dart isolate has a single thread of execution and
-shares no mutable objects with other isolates.
-To communicate with each other,
-isolates use message passing.
-Many Dart apps use only one isolate, the _main isolate_.
-You can create additional isolates to enable
-parallel code execution on multiple processor cores.
-
-Although Dart's isolate model is built with underlying primitives
-such as processes and threads
-that the operating system provides,
-the Dart VM's use of these primitives
-is an implementation detail that this page doesn't discuss.
+Concurrent programming in Dart refers to both asynchronous APIs, like `Future`
+and `Stream`, and `isolates`, which allow you to move processes to separate
+cores. All Dart code runs in isolates.
+
+An isolate is like a small space on your machine that has its own isolated
+memory. If you’re coming from another language that supports multiple threads,
+you’ll find that isolates are similar, but are crucially different in that
+isolates can’t share memory. Most Dart apps run all their code in a single
+isolate, the main isolate, but you can create more isolates if you need them.
+When you spawn a new isolate, it will have its own isolated memory heap, and it
+will have its own event loop. The event loop is what makes asynchronous and
+concurrent programming possible in Dart.
+
+## Event Loop
+
+Dart’s runtime model is based on an event loop. This event loop is responsible
+for executing the code of your program, collecting and processing events,
+running garbage collection processes, and more. As your application runs, all
+events are added to a queue, called the event queue. Events can be anything from
+requests to repaint the UI, to user taps and keystrokes, to i/o from the disk.
+Because your app can’t predict what order these events will happen, it runs this
+event loop that processes events in the order they're queued, one at a time.
```

Comment:
```suggestion
Dart’s runtime model is based on an event loop. 
The event loop is responsible for executing your program's code,
collecting and processing events, running garbage collection processes, and more.

As your application runs, all events are added to a queue,
called the *event queue*. Events can be anything from requests to repaint the UI,
to user taps and keystrokes, to I/O from the disk.
Because your app can’t predict what order events will happen, 
the event loop processes events in the order they're queued, one at a time.
```
(also just a minor restructure)